src/test/datascience/data-viewing/dataViewerDependencyService.unit.test.ts

```diff
@@ -4,118 +4,136 @@
+
     test('All ok, if pandas is installed and version is > 1.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.30.0' });
-        await dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const version = '3.3.3';
+
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
+
+        const result = await dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+        assert.equal(result, undefined);
+        assert.deepEqual(
+            stub.getCalls().map((call) => call.lastArg),
+            [getVersionOfPandasCommand]
+        );
     });
+
+    test('All ok, if pandas is installed and version is > 1.20, even if the command returns with a new line', async () => {
+        const version = '1.4.2\n';
+
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
+
+        const result = await dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+        assert.equal(result, undefined);
+        assert.deepEqual(
+            stub.getCalls().map((call) => call.lastArg),
+            [getVersionOfPandasCommand]
+        );
+    });
+
     test('Throw exception if pandas is installed and version is = 0.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.20.0' });
+        const version = '0.20.0';
 
-        const promise = dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const stub = sinon.stub(helpers, 'executeSilently');
+        stub.returns(Promise.resolve([{ ename: 'stdout', output_type: 'stream', text: version }]));
 
-        await assert.isRejected(promise, DataScience.pandasTooOldForViewingFormat().format('0.20.'));
+        const resultPromise = dependencyService.checkAndInstallMissingDependenciesOnKernel(kernel);
+        await assert.isRejected(
+            resultPromise,
+            DataScience.pandasTooOldForViewingFormat().format('0.20.'),
+            'Failed to identify too old pandas'
+        );
+        assert.deepEqual(
+            stub.getCalls().map((call) => call.lastArg),
+            [getVersionOfPandasCommand]
+        );
     });
+
     test('Throw exception if pandas is installed and version is < 0.20', async () => {
-        when(
-            pythonExecService.exec(deepEqual(['-c', 'import pandas;print(pandas.__version__)']), anything())
-        ).thenResolve({ stdout: '0.10.0' });
+        const version = '0.10.0';
 
-        const promise = dependencyService.checkAndInstallMissingDependencies(interpreter);
+        const stub = sinon.stub(helpers, 'executeSilently');
```

Comment:
not sure if this works, however ifeel it might be easier to stub the `kernel. session` methods.

Reply:
If it isn't working I'm hallucinating 😅 I wouldn't discard that possibility though